node/index.js

```diff
@@ -127,6 +129,18 @@ function TChannel(options) {
 
             self.logger.info(self.hostPort + ' listening');
             self.emit('listening');
+
+            // TODO: make this in constructor and not listening event
+            self.tracer = new TracingAgent({
```

Comment:
At least initialize self.tracer to null during construction, the definition of a new field eventually will cause a change of type for self currently.

Even better if agent could support having host/port optionally filled in later.

Note that ephemeral clients, like TCurl, don't necessarily listen on a socket at all.